python_modules/dagster/dagster/core/definitions/unresolved_asset_job_definition.py

```diff
@@ -78,22 +107,43 @@ def _selection_from_string(string: str) -> "AssetSelection":
 def define_asset_job(
     name: str,
     selection: Optional[Union[str, Sequence[str], "AssetSelection"]] = None,
+    config: Optional[Union[ConfigMapping, Dict[str, Any], "PartitionedConfig"]] = None,
     description: Optional[str] = None,
     tags: Optional[Dict[str, Any]] = None,
+    partitions_def: Optional["PartitionsDefinition"] = None,
 ) -> UnresolvedAssetJobDefinition:
     """Creates a definition of a job which will materialize a selection of assets. This will only be
     resolved to a JobDefinition once placed in a repository.
 
     Args:
         name (str):
-            The name for the JobDefintion.
+            The name for the Job.
         selection (Union[str, Sequence[str], AssetSelection]):
             A selection over the set of Assets available on your repository. This can be a string
             such as "my_asset*", a list of such strings (representing a union of these selections),
             or an AssetSelection object.
 
             This selection will be resolved to a set of Assets once the repository is loaded with a
             set of AssetsDefinitions.
+        config:
+            Describes how the Job is parameterized at runtime.
+
+            If no value is provided, then the schema for the job's run config is a standard
+            format based on its solids and resources.
+
+            If a dictionary is provided, then it must conform to the standard config schema, and
+            it will be used as the job's run config for the job whenever the job is executed.
+            The values provided will be viewable and editable in the Dagit playground, so be
+            careful with secrets.
+
+            If a :py:class:`ConfigMapping` object is provided, then the schema for the job's run config is
+            determined by the config mapping, and the ConfigMapping, which should return
+            configuration in the standard format to configure the job.
+
+            If a :py:class:`PartitionedConfig` object is provided, then it defines a discrete set of config
+            values that can parameterize the job, as well as a function for mapping those
+            values to the base config. The values provided will be viewable and editable in the
+            Dagit playground, so be careful with secrets.
```

Comment:
Worth adding docstring for `partitions_def`. And, in it, documenting any constraints (all assets selected in the job need to have that partitioning?)